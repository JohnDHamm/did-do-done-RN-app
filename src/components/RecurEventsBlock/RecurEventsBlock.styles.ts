import styled from 'styled-components/native';
import { FONTS } from '../../styles';

export const Block = styled.View`
  align-items: center;
`;
export const Header = styled.Text<Theme>`
  font-family: ${FONTS.PRIMARY};
  font-size: 20px;
  color: ${(props) => props.theme.purple};
`;

export const MinRow = styled.View`
  flex-direction: row;
  align-items: center;
  padding-top: 10px;
`;

export const BadgeContainer = styled.View`
  padding-left: 10px;
`;

export const BadgeLabel = styled.Text<{ color: string }>`
  color: ${(props) => props.color};
  font-family: ${FONTS.PRIMARY};
  font-size: 22px;
  padding-left: 8px;
`;
