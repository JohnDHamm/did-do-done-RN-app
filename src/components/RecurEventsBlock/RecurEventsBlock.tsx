import React from 'react';
import {
  BadgeContainer,
  BadgeLabel,
  Block,
  Header,
  MinRow,
} from './RecurEventsBlock.styles';
import RecurNumberBadge from '../RecurNumberBadge/RecurNumberBadge';
import { COLORS } from '../../styles';

interface BlockProps {
  recurTotals: RecurTotals;
  minimized?: boolean;
}

const RecurEventsBlock: React.FC<BlockProps> = ({
  recurTotals,
  minimized = false,
}) => {
  const renderMinRecurBadges = () => {
    const badgeTypes: Array<RecurType> = ['missed', 'today', 'thisweek'];
    return badgeTypes.map((badgeType) => {
      if (recurTotals[badgeType] > 0)
        return (
          <BadgeContainer key={badgeType}>
            <RecurNumberBadge type={badgeType} total={recurTotals[badgeType]} />
          </BadgeContainer>
        );
    });
  };

  return !minimized ? (
    <Block>
      <Header>do again:</Header>
      {recurTotals.missed > 0 && (
        <MinRow>
          <RecurNumberBadge type="missed" total={recurTotals.missed} />
          <BadgeLabel color={COLORS.MISSED_RED}>missed</BadgeLabel>
        </MinRow>
      )}
      {recurTotals.today > 0 && (
        <MinRow>
          <RecurNumberBadge type="today" total={recurTotals.today} />
          <BadgeLabel color={COLORS.TODAY_GREEN}>today</BadgeLabel>
        </MinRow>
      )}
      {recurTotals.thisweek > 0 && (
        <MinRow>
          <RecurNumberBadge type="thisweek" total={recurTotals.thisweek} />
          <BadgeLabel color={COLORS.THIS_WEEK_YELLOW}>this week</BadgeLabel>
        </MinRow>
      )}
      <MinRow>
        <RecurNumberBadge type="next30" total={recurTotals.next30} />
        <BadgeLabel color={COLORS.PRIMARY_GRAY}>next 30 days</BadgeLabel>
      </MinRow>
    </Block>
  ) : (
    <MinRow>
      <Header>do again:</Header>
      {renderMinRecurBadges()}
      <BadgeContainer key={'next30'}>
        <RecurNumberBadge type={'next30'} total={recurTotals.next30} />
      </BadgeContainer>
    </MinRow>
  );
};

export default RecurEventsBlock;
