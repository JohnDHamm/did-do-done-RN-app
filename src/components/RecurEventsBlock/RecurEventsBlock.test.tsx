import React from 'react';
import { cleanup } from 'react-native-testing-library';
import RecurEventsBlock from './RecurEventsBlock';
import { renderWithLightTheme } from '../../testing/helpers';

const recurTotals: RecurTotals = {
  missed: 1,
  today: 2,
  thisweek: 4,
  next30: 3,
};

describe('RecurEventsBlock', () => {
  afterEach(cleanup);

  it('renders correctly with defaults', () => {
    const testRenderer = renderWithLightTheme(
      <RecurEventsBlock recurTotals={recurTotals} />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly with minimized prop set to true', () => {
    const testRenderer = renderWithLightTheme(
      <RecurEventsBlock recurTotals={recurTotals} minimized />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});
