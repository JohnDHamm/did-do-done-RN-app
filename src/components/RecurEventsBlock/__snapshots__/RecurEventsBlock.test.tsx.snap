// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurEventsBlock renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#4c408f",
          "fontFamily": "Lobster-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    do again:
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      color="#B70000"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#B70000",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <Text
      color="#B70000"
      style={
        Array [
          Object {
            "color": "#B70000",
            "fontFamily": "Lobster-Regular",
            "fontSize": 22,
            "paddingLeft": 8,
          },
        ]
      }
    >
      missed
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      color="#237b59"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#237b59",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <Text
      color="#237b59"
      style={
        Array [
          Object {
            "color": "#237b59",
            "fontFamily": "Lobster-Regular",
            "fontSize": 22,
            "paddingLeft": 8,
          },
        ]
      }
    >
      today
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      color="#ffd764"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffd764",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        4
      </Text>
    </View>
    <Text
      color="#ffd764"
      style={
        Array [
          Object {
            "color": "#ffd764",
            "fontFamily": "Lobster-Regular",
            "fontSize": 22,
            "paddingLeft": 8,
          },
        ]
      }
    >
      this week
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      color="#c3c3c3"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#c3c3c3",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        3
      </Text>
    </View>
    <Text
      color="#c3c3c3"
      style={
        Array [
          Object {
            "color": "#c3c3c3",
            "fontFamily": "Lobster-Regular",
            "fontSize": 22,
            "paddingLeft": 8,
          },
        ]
      }
    >
      next 30 days
    </Text>
  </View>
</View>
`;

exports[`RecurEventsBlock renders correctly with minimized prop set to true 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingTop": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#4c408f",
          "fontFamily": "Lobster-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    do again:
  </Text>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 10,
        },
      ]
    }
  >
    <View
      color="#B70000"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#B70000",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 10,
        },
      ]
    }
  >
    <View
      color="#237b59"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#237b59",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        2
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 10,
        },
      ]
    }
  >
    <View
      color="#ffd764"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffd764",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        4
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 10,
        },
      ]
    }
  >
    <View
      color="#c3c3c3"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#c3c3c3",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Lobster-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        3
      </Text>
    </View>
  </View>
</View>
`;
